pettingzoo/classic/tictactoe/tictactoe.py

```diff
@@ -56,7 +59,18 @@ def observe(self, agent):
 
         cur_p_board = np.equal(board_vals, cur_player + 1)
         opp_p_board = np.equal(board_vals, opp_player + 1)
-        return np.stack([cur_p_board, opp_p_board], axis=2).astype(np.int8)
+
+        observation = np.stack([cur_p_board, opp_p_board], axis=2).astype(np.int8)
+
+        legal_moves = self.infos[agent]['legal_moves']
+        action_mask = np.zeros(9, int)
+        for i in legal_moves:
+            action_mask[i] = 1
+
+        self.observation_spaces[agent]['observation'] = observation
```

Comment:
The observation space should not be changed after the environment is constructed. So these lines should not be there.